File: src/Portfolio/Portfolio.Blazor/Data/Entities/CreditEntry.cs
using System.ComponentModel.DataAnnotations;

namespace Portfolio.Blazor.Data.Entities;

public sealed class CreditEntry
{
    public int Id { get; set; }
    public string? Url { get; set; }

    [Required(ErrorMessage = "Category is required.")]
    public string Category { get; set; } = string.Empty;
    public int Order { get; set; }
    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public bool IsAutomatic { get; set; }
    public bool Active { get; set; }
    public string? PackageManager { get; set; }
    public string? PackageName { get; set; }

    public List<CreditEntryTranslation> Translations { get; set; } = [];
}

File: laravel/app/Models/CreditEntry.php
<?php

namespace App\Models;

use App\Models\Concerns\Auditable;
use App\Models\Concerns\HasTranslations;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

/**
 * @method CreditEntryTranslation|null translation(?string $locale = null)
 */
class CreditEntry extends Model
{
    use Auditable, HasFactory, HasTranslations;

    protected $fillable = [
        'url', 'category', 'order',
        'is_automatic', 'active', 'package_manager', 'package_name',
    ];

    protected $casts = [
        'is_automatic' => 'boolean',
        'active' => 'boolean',
    ];

    /**
     * @return HasMany<CreditEntryTranslation, $this>
     */
    public function translations(): HasMany
    {
        return $this->hasMany(CreditEntryTranslation::class);
    }
}
